tests/sentry/middleware/test_ratelimit_middleware.py

```diff
@@ -215,3 +212,58 @@ class TestEndpoint:
             pass
 
         assert get_rate_limit_value("GET", TestEndpoint, "ip") is None
+
+
+class RateLimitHeaderTestEndpoint(Endpoint):
+    permission_classes = (AllowAny,)
+
+    enforce_rate_limit = True
+    rate_limits = {"GET": {RateLimitCategory.IP: RateLimit(2, 100)}}
```

Comment:
It was primarily to take advantage to `dict.get(key, default)` so that child endpoints can override individual rate limits without having to copy the entire thing.

I suppose this could be done with getattr.

Let's discuss this outside this PR.

Reply:
Oh yeah, also I can directly plug in the `request.META["HTTP_METHOD"]` as the key to get the rate limit